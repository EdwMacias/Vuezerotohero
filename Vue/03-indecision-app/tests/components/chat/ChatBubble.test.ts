import { mount } from '@vue/test-utils';
import ChatBubble from '@/components/chat/ChatBubble.vue';

describe('<ChatBubble/>', () => {
  test('render own message correctly', () => {
    const message = 'Hola mundo';
    const wrapper = mount(ChatBubble, {
      props: {
        message,
        itsMine: true,
      },
    });
    expect(wrapper.find('.bg-blue-200').exists()).toBe(true);
    expect(wrapper.find('.bg-blue-200').text()).toContain(message);
    expect(wrapper.find('.bg-gray-200').exists()).toBeFalsy();
  });

  test('render received message correctly', () => {
    const message = 'Hola mundo';
    const wrapper = mount(ChatBubble, {
      props: {
        message,
        itsMine: false,
      },
    });
    expect(wrapper.find('.bg-gray-300').exists()).toBe(true);
    expect(wrapper.find('.bg-blue-200').exists()).toBe(false);
    expect(wrapper.find('img').exists()).toBe(false);
    console.log('no image');
  });
  test('render received message correctly with image', () => {
    const message = 'Hola mundo';
    const image = 'example.jpg';
    const wrapper = mount(ChatBubble, {
      props: {
        message,
        itsMine: false,
        image,
      },
    });
    expect(wrapper.find('.bg-gray-300').exists()).toBe(true);
    expect(wrapper.find('.bg-blue-200').exists()).toBe(false);
    expect(wrapper.find('img').attributes('src')).toBe(image);
  });
});
