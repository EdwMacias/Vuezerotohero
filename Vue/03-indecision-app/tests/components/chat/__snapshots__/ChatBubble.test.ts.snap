// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ChatBubble/> > render own message correctly 1`] = `
"<!-- Example Received Message -->
<div class="flex">
  <div class="bg-gray-300 text-black p-2 rounded-lg max-w-xs">Hola mundo
    <!--v-if-->
  </div>
</div>"
`;
